### PR TITLE
fix(operator): fix snitch deployment for openshift

### PR DESCRIPTION
**Pull request by rksharma95**
**Purpose of PR?**:
For some unknown issues the [`bpflsmprobe`](https://github.com/kubearmor/KubeArmor/tree/main/KubeArmor/utils/bpflsmprobe) utility is not working without `hostPID: true`. As a temporary fix this PR set hostPID to true when the operator get deployed on the OpenShift Platform. 

**Does this PR introduce a breaking change?**
No
**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [x] Bug fix. Fixes #<issue number> **(No Issue Opened for it )**
- [x] New feature (non-breaking change which adds functionality) **(Not Applicable)**
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected) **(No)**
- [x] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [x] Commit has unit tests **(Not Applicable)**
- [x] Commit has integration tests **(Not Applicable)**

### CODE REVIEW

File: pkg/KubeArmorOperator/Dockerfile [hunk 1]
```diff
@@ -37,7 +37,29 @@ COPY $OPERATOR_DIR/runtime runtime
...
+      vendor="Accuknox" \
+      version=${VERSION} \
+      release=${VERSION} \
+      summary="kubearmor-operator container image based on redhat ubi" \
+      description="kubearmor-operator to deploy and manage KubeArmor"
+
+RUN microdnf -y update && \
+    microdnf -y install --nodocs --setopt=install_weak_deps=0 --setopt=keepcache=0 shadow-utils libcap && \
+    microdnf clean all && \
+    rm -rf /var/cache/yum
```
Comment (nyrahul) [inline pkg/KubeArmorOperator/Dockerfile]: Shouldn't the folder name be dnf?
```suggestion
    rm -rf /var/cache/dnf
```

**Review by nyrahul — APPROVED** [review]
Mostly LGTM. Please check one review comment inline.

File: pkg/KubeArmorOperator/Dockerfile (hunk 1 above)
Comment (rksharma95) [reply pkg/KubeArmorOperator/Dockerfile]: microdnf keep cache in `/var/cache/yum`. we can also remove this as cache will be removed by `microdnf clean all`.

**Comment by rksharma95** [comment]
Related: #1200

**Pull request merged**